File: java/app/src/main/java/com/pgssoft/mvvm/viewmodels/interfaces/IMainActivityAccess.java
package com.pgssoft.mvvm.viewmodels.interfaces;

import com.pgssoft.mvvm.model.database.Rate;
import com.pgssoft.mvvm.views.adapters.interfaces.handlers.RateItemHandler;
import com.pgssoft.mvvm.views.adapters.interfaces.providers.BaseItemProvider;

/**
 * Created by bstokrocki on 30.01.2017.
 */
public interface IMainActivityAccess extends IBaseActivityAccess {
    void displayRates(BaseItemProvider<Rate> ratesProvider, RateItemHandler rateItemHandler);

    void openRateDetailsScreen(Rate rate);
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/viewmodel/iface/IMainActivityAccess.kt
package com.pgssoft.kotlinplayground.viewmodel.iface

import com.pgssoft.kotlinplayground.model.db.Rate
import com.pgssoft.kotlinplayground.view.adapter.providers.BaseItemProvider

/**
 * Created by bstokrocki on 30.01.2017.
 */
interface IMainActivityAccess : IBaseActivityAccess {
    fun displayRates(ratesProvider: BaseItemProvider<Rate>)

    fun openRateDetailsScreen(rate: Rate)
}
